genie-web/src/main/java/com/netflix/genie/web/tasks/leader/UserMetricsTask.java

```diff
@@ -88,37 +92,77 @@ public long getFixedRate() {
      */
     @Override
     public void run() {
+        log.info("Publishing user metrics");
 
         final Map<String, UserResourcesSummary> summaries = this.jobSearchService.getUserResourcesSummaries();
 
-        // Set gagues back to 0 for all users whose metric were published the previous iterations.
-        for (final String user : this.usersToReset) {
-            // Unless we're about to publish new values, in which case don't bother.
-            if (!summaries.containsKey(user)) {
-                this.publishUserMetrics(user, 0, 0);
-            }
-        }
+        // Update number of active users
+        log.info("Number of users with active jobs: {}", summaries.size());
+        this.activeUsersCount.set(summaries.size());
+
+        // Track users who previously had jobs but no longer do
+        final Set<String> usersToReset = Sets.newHashSet(this.userResourcesRecordMap.keySet());
```

Comment:
Can use Sets.difference(this.userResourcesRecordMap.keySet(),summaries.keySet())

Reply:
There are a couple of ways, such as `Sets.difference` that do look better in terms of readability. However they produce views of the sets and then throw `ConcurrentModificationException` when I'm iterating this difference and removing from the map.